### DISCUSSION TITLE
Adding heatmaps

### DISCUSSION THREAD

**Discussion by mmfs37** [Richardsl/heatmap-calendar-obsidian#5] [Q&A] [answered]
I am very new to github, obsidian and this add on so maybe my question is stupid.
I used this addon for habit tracking on my daily page. But bc of tracking about 15 habits and having everytime the heatmap and then [habit::True] was too much.
I have now a page where I can see every data heatmap of my habits, but I do not have it in my daily page. 
Having it in my daily page motivated me so I got a smart Idea.
**What if I get every completet habit one point, and then those points added get you the daily datapoint for the heatmap?**

So I would use it something like this:

[Meditate::True] 
[Reading::True] 
[Workout::True] 
[EatHealthy::False] 
<- for this example date the heatmap would have the colour intensity of 3
if I would only check 2 off I would have 2 as intensity..

Is that in any way possible to do?

**Comment by Richardsl** [comment, marked as answer]
yes should be able with some javascript.
but this plugin is just about displaying the data, not collecting or manipulating the data

**Reply by mmfs37** [reply]
is there any tutorial or reddit post or anything that explains that further? I would be very interested how that could be done automatically in obsidian.

**Reply by sfdrada** [reply]
Try this out


```dataviewjs
dv.span("**Heatmap 🚥**")

const calendarData = {
    year: 2022,
    colors: {
	    scale:
["#FF4D4D","#FF884D","#FFFF4D","#6AFF4D"] // red,orange,yellow,green
    },
	intensityScaleStart: 1,
    intensityScaleEnd: 4,
    entries: []
}

for(let page of dv.pages('"Journals 📜"').where(p=>p.Test1).sort(p=>p.file.name)){

	// Change the habits you want to add here
	// Converts booleans into 1 (true) or 0 (false) and adds them
	page.HabitSum = dv.tryEvaluate ("Meditate + Reading + Workout + EatHealthy", {Meditate:dv.compare(page.Meditate,false), Reading:dv.compare(page.Reading,false), Workout:dv.compare(page.Workout,false), EatHealthy:dv.compare(page.EatHealthy,false)})


    calendarData.entries.push({
        date: page.file.name,
        intensity: page.HabitSum,
		color: "scale", 
    })
       
}

renderHeatmapCalendar(this.container, calendarData)
```

**Reply by maifather** [reply]
Hi thank you so much. I tried it out but they display nothing onto the heatmap.
Here's my code:
<img width="372" alt="image" src="https://github.com/Richardsl/heatmap-calendar-obsidian/assets/139601249/bc1c0576-28ed-41fc-912d-99d99539061f">
<img width="205" alt="image" src="https://github.com/Richardsl/heatmap-calendar-obsidian/assets/139601249/b01e769b-0fec-4335-9cea-b7bba14cede4">

1. What's your "Test1" property?
2. And what type properties Meditate, Reading, etc. are? Are they number type (so you can type 1,0) or checkbox type?

I don't know coding so my question might sound dumb but please help me. I really thank you

**Reply by sfdrada** [reply]
I recommend asking chatgpt to help you understand the code and set it up how you want since i have long since stopped using heatmaps. 

However, I remember it being a boolean like it would be metadata at the top of the file or within the obsidian file like Meditate:: False and then combining it with Better Inline Fields plugin to convert the inline metadata True/False boolean into a checkbox so that it would be easy to change and it updated to the heatmap, keep in mind this was before Obsidian updated the metadata to be able just click the checkbox if it were at the top of the file.

Good luck, hope you figure it out and post it below!

PS. Rereading it i think it was referencing the folder which was my journals folder and the Test1 was the filename

PPS. Good luck learning how to code :)
